### PR TITLE
Image Pull

### PR DESCRIPTION
**Pull request by brookeclouston**
Updated original script to search for images by string of catalog number, not by integer of catalog number. Updated original to include an argument for the user to enter metadata field names when searching by metadata rather then changing a hard coded dictionary. 
Updated test script to account for these changes.

### CODE REVIEW

File: ImagePull.py [hunk 1]
```diff
@@ -1,192 +1,124 @@
...
-
-# searches source directory list for file names contained by the operator passed in for the single file name entered
-def single_file_search(image, op,source):
-    return [file for file in source_search(source) if op(configure(os.path.basename(file)), int(image))]
-
-# searches source list for file names in range of file names entered
-def range_file_search(image1, image2,source):
-    return [file for file in source_search(source) if
-            ge(configure(os.path.basename(file)), int(image1)) and le(configure(os.path.basename(file)), int(image2))]
+    # Returns a list of each .JPG file path in the source directory
```
Comment (kcranston) [inline ImagePull.py]: do we know for sure that all of the image files have extension uppercase .JPG (as opposed to .jpg or .jpeg or other extensions?)

Comment (brookeclouston) [reply ImagePull.py]: They will be of "jpeg" type, I have updated to flag files with endings as `.JPG/.jpg/.JPEG/.jpeg`

**Comment by kcranston** [comment]
I am getting a warning when running the tests.

```
(env) $  specify-scripts git:(image_pull) ✗ python test_ImagePull.py
...csv file not found, check file path
./Users/karen/Documents/aafc/specify-scripts/ImagePull.py:48: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/karen/Documents/aafc/specify-scripts/testcsv.csv' mode='r' encoding='UTF-8'>
  csv_list = [row[0] for row in csv.reader(open(csv_file, newline=""))]
.....
----------------------------------------------------------------------
Ran 9 tests in 0.065s

OK
```
Opening the files with `with` will ensure they get closed: see [IO docs](https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files)

**Comment by brookeclouston** [comment]
Script has been updated so that `with` method is used and this warning no longer appears

**Comment by kcranston** [comment]
Ok, working now. Thanks!

**Pull request merged**